api/domain/v2beta1/domain_utils.go

```diff
@@ -135,3 +136,214 @@ func (s *API) WaitForDNSRecordExist(
+				Domain: req.Domain,
+			}, opts...)
+			if err != nil {
+				return nil, false, err
+			}
+
+			lastStatus = resp.Status
+
+			// Terminal success statuses
+			if _, isTerminal := terminalStatuses[resp.Status]; isTerminal {
+				return resp, true, nil
+			}
+
+			// Terminal error statuses
+			if _, isError := errorStatuses[resp.Status]; isError {
+				return nil, true, errors.New("domain entered error status: " + string(resp.Status))
+			}
+
+			// Transient statuses
+			if _, isTransient := transientStatuses[resp.Status]; isTransient {
+				return resp, false, nil
+			}
+
+			// Unexpected status
+			return nil, false, errors.New("unexpected domain status: " + string(resp.Status))
+		},
+		Timeout:          timeout,
+		IntervalStrategy: async.LinearIntervalStrategy(retryInterval),
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "waiting for domain %s failed, last known status: %s", req.Domain, lastStatus)
+	}
+
+	return domain.(*Domain), nil
+}
+
+// WaitForAutoRenewStatusRequest defines the parameters for waiting on the auto‑renew feature.
+type WaitForAutoRenewStatusRequest struct {
+	Domain        string         // The domain to wait for.
+	Timeout       *time.Duration // Optional timeout.
+	RetryInterval *time.Duration // Optional retry interval.
+}
+
+// WaitForAutoRenewStatus polls the domain until its auto‑renew feature reaches a terminal state
+// (either "enabled" or "disabled"). It uses GetDomain() to fetch the current status.
+func (s *RegistrarAPI) WaitForAutoRenewStatus(req *WaitForAutoRenewStatusRequest, opts ...scw.RequestOption) (*Domain, error) {
+	// Use default timeout and retry interval if not provided.
+	timeout := defaultTimeout
+	if req.Timeout != nil {
+		timeout = *req.Timeout
+	}
+	retryInterval := defaultRetryInterval
+	if req.RetryInterval != nil {
+		retryInterval = *req.RetryInterval
+	}
+
+	// Terminal statuses for auto_renew: enabled or disabled.
+	terminalStatuses := map[DomainFeatureStatus]struct{}{
+		DomainFeatureStatusEnabled:  {},
+		DomainFeatureStatusDisabled: {},
+	}
+	// Transient statuses for auto_renew: enabling or disabling.
+	transientStatuses := map[DomainFeatureStatus]struct{}{
+		DomainFeatureStatusEnabling:  {},
+		DomainFeatureStatusDisabling: {},
+	}
+
+	var lastStatus DomainFeatureStatus
+
+	domainResult, err := async.WaitSync(&async.WaitSyncConfig{
+		Get: func() (interface{}, bool, error) {
+			// Retrieve the domain.
+			resp, err := s.GetDomain(&RegistrarAPIGetDomainRequest{
+				Domain: req.Domain,
+			}, opts...)
+			if err != nil {
+				return nil, false, err
+			}
+
+			// Check the current auto_renew status.
+			lastStatus = resp.AutoRenewStatus
+			if _, isTerminal := terminalStatuses[resp.AutoRenewStatus]; isTerminal {
+				return resp, true, nil
+			}
+			if _, isTransient := transientStatuses[resp.AutoRenewStatus]; isTransient {
+				return resp, false, nil
+			}
+			return nil, false, errors.New("unexpected auto_renew status: " + string(resp.AutoRenewStatus))
```

Comment:
Regardless of the transient state, it should continue if no terminal status is encountered. A new transient state in the API should not make the waiter fail.